vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -383,14 +383,29 @@ func (r *RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver)
 			return err
 		}
 		log.Debugf("  Mounts for %v", ct.Names)
-
 		for _, mount := range containerJSONInfo.Mounts {
-			// check if the mount location belongs to vsphere plugin
-			if matchNameforVMDK(mount.Source) {
-				r.Incr(mount.Name)
-				log.Infof("  name=%v (driver=%s source=%s) (%v)",
-					mount.Name, mount.Driver, mount.Source, mount)
+			// check if the mount location belongs to vmdk plugin
+			if isVMDKMount(mount.Source) != true {
+				continue
+			}
+			volname := mount.Name
+			// gets hit once to retrieve the default datastore. datastoreName is reused henceforth
+			if datastoreName == "" {
```

Comment:
this is done once,  but needs to be done inside the loop since we need a volume name to get the datastore.